scripts/wallet.js

```diff
@@ -281,16 +298,16 @@ export class Wallet {
     /**
      * @return [string, string] Address and its BIP32 derivation path
      */
-    getNewAddress() {
-        const last = this.#highestUsedIndex;
+    getNewAddress(nReceiving = 0) {
+        const last = this.#highestUsedIndices.get(nReceiving);
         this.#addressIndex =
             (this.#addressIndex > last ? this.#addressIndex : last) + 1;
         if (this.#addressIndex - last > MAX_ACCOUNT_GAP) {
             // If the user creates more than ${MAX_ACCOUNT_GAP} empty wallets we will not be able to sync them!
             this.#addressIndex = last;
         }
-        const path = this.getDerivationPath(0, this.#addressIndex);
-        const address = this.getAddress(0, this.#addressIndex);
+        const path = this.getDerivationPath(nReceiving, this.#addressIndex);
```

Comment:
the variable #addressIndex should also distinguish between the two chains  and so you should make it a map